File: frontend/src/microservices/order/objects/PayOrderPlannerRequest.ts
// 本文件定义订单支付请求，和后端 `PayOrderPlannerRequest` 保持同名镜像。

export type PayOrderPlannerRequest = {
  orderId: string
  paymentMethod: string
  paymentSucceeded: boolean
  travelerIds?: string[] | null
}

export const payOrderPlannerRequestFromJson = (json: string): PayOrderPlannerRequest =>
  JSON.parse(json) as PayOrderPlannerRequest

export const payOrderPlannerRequestToJson = (value: PayOrderPlannerRequest): string =>
  JSON.stringify(value)

File: backend/src/main/scala/microservices/order/objects/PayOrderPlannerRequest.scala
package com.typesafe.travel.order.domain

import io.circe.{Decoder, Encoder}
import io.circe.generic.semiauto.{deriveDecoder, deriveEncoder}

final case class PayOrderPlannerRequest(orderId: String, paymentMethod: String, paymentSucceeded: Boolean, travelerIds: Option[List[String]] = None)
object PayOrderPlannerRequest:
  given Encoder[PayOrderPlannerRequest] = deriveEncoder
  given Decoder[PayOrderPlannerRequest] = deriveDecoder
